ondewo/t2s/text-to-speech.proto

```diff
@@ -19,34 +19,107 @@ import "google/protobuf/empty.proto";
     rpc Synthesize (SynthesizeRequest) returns (SynthesizeResponse) {
     };
+    // will this safe time when doing batch predict on the AI model?
+    rpc BatchSynthesize (BatchSynthesizeRequest) returns (BatchSynthesizeResponse) {
+    };
+    // Retrieves the configuration of the specified pipeline.
     rpc GetT2sPipeline (T2sPipelineId) returns (Text2SpeechConfig) {
     };
+    // Creates a pipeline with the specified configuration and retrieves its id.
     rpc CreateT2sPipeline (Text2SpeechConfig) returns (T2sPipelineId) {
     };
+    // Deletes the specified pipeline.
     rpc DeleteT2sPipeline (T2sPipelineId) returns (google.protobuf.Empty) {
     };
+    // Update a specified pipeline with certain configuration.
     rpc UpdateT2sPipeline (Text2SpeechConfig) returns (google.protobuf.Empty) {
     };
+    // Retrieve the list of pipelines with an specific requirement.
     rpc ListT2sPipelines (ListT2sPipelinesRequest) returns (ListT2sPipelinesResponse) {
     };
+    // Retrieve the list of languages given a specific config request.
     rpc ListT2sLanguages (ListT2sLanguagesRequest) returns (ListT2sLanguagesResponse) {
     };
+    // Retrieve the list of domains given a specific config request.
     rpc ListT2sDomains (ListT2sDomainsRequest) returns (ListT2sDomainsResponse) {
     };
 }
 
 ////////////////
 // SYNTHESIZE //
 ////////////////
+
+// Represents a Synthesize Request.
+// A Synthesize Request contains the information need to perform a text to speech conversion.
 message SynthesizeRequest {
-    string t2s_pipeline_id = 1; // id of the model that will generate audio
-    string text = 2; // text for t2s
-    float length_scale = 3; // this parameter defines the speed of the audio. Should not be very much off 1.0
-    float noise_scale = 4; // can be from 0.0 to 1. Defines the noise in the generated audio
-    int32 sample_rate = 5; // by default 22050 sample rate of the generated wav file
-    enum Pcm {
+
+    // Required. Represents the text that will be transformed to speech.
+    string text = 1;
+
+    // Required. Represents the specifications needed to do the text to speech transformation.
+    RequestConfig config = 2;
+}
+
+message BatchSynthesizeRequest {
+
+    repeated string text = 1;
+
+    // Required. Represents the specifications needed to do the text to speech transformation.
+    repeated RequestConfig config = 2;
+}
+
+message BatchSynthesizeResponse {
+
+    repeated SynthesizeResponse response = 1;
+
+}
+
+// Represents a Configuration for the text to speech conversion.
+message RequestConfig {
+
+    // Required. Represents the pipeline id of the model configuration that will be used.
+    string t2s_pipeline_id = 1;
+
+    // Optional. This parameter is used for time stretching which is the process of
+    // changing the speed or duration of an audio.
+    // It should be much more than 1.0. O is not a valid number for this variable.
+    // The default value is 1.
```

Comment:
There seems to be a small problem here: You say that this field is optional, and that its default value is 1. If a field is optional, it's up to you to provide it or not - but if you don't explicitly set it here, it will just take the default value for a grpc proto float which is 0.